File: scripts/test_solutions/007_reverse_integer.cpp
class Solution {
public:
    int reverse(int x) {
        long result = 0;
        while (x != 0) {
            result = result * 10 + x % 10;
            x /= 10;
        }
        if (result < INT_MIN || result > INT_MAX) return 0;
        return (int)result;
    }
};

File: scripts/test_solutions/007_reverse_integer.js
var reverse = function(x) {
    const sign = x < 0 ? -1 : 1;
    let result = 0;
    x = Math.abs(x);
    while (x > 0) {
        result = result * 10 + (x % 10);
        x = Math.floor(x / 10);
    }
    result *= sign;
    if (result < -(2**31) || result > 2**31 - 1) return 0;
    return result;
};
